pkg/config/config_test.go

```diff
@@ -473,6 +500,58 @@ func (s *MySuite) TestAddKindToModules(c *C) {
 	c.Assert(testMod.Kind, Equals, expected)
 }
 
+func (s *MySuite) TestModuleConnections(c *C) {
+	dc := getMultiGroupDeploymentConfig()
+	modID0 := dc.Config.DeploymentGroups[0].Modules[0].ID
+	modID1 := dc.Config.DeploymentGroups[1].Modules[0].ID
+
+	dc.addSettingsToModules()
+	dc.addMetadataToModules()
+	dc.addDefaultValidators()
+
+	err := dc.applyUseModules()
+	c.Assert(err, IsNil)
+	err = dc.applyGlobalVariables()
+	c.Assert(err, IsNil)
+	err = dc.expandVariables()
+	// TODO: this will become nil once intergroup references are enabled
+	c.Assert(err, NotNil)
+
+	// check that ModuleConnections has map keys for each module ID
+	c.Assert(len(dc.moduleConnections), Equals, 2)
```

Comment:
Can we use DeepEquals instead of point-checks?
Say `dc.moduleConnections` contains just one entry.
Your test will fails with `1 != 2` and no additional information.
But if we write it like this:
```
c.Check(dc.moduleConections, DeepEquals, map[string][]ModConnection{
  modID0: []ModConnection{
    {kind: deploymentConnection, sharedVariables: []string{"project_id"}, ...etc},
    {kind: deploymentConnection, sharedVariables: []string{"deployment_name"}, ...etc},
  }
  modID1: []ModConnection{...etc}
})
```
The error message will show you exactly what went wrong.
Pros:
* 1 check instead of dozen
* shows all information
* better test readability
Cons: 
* rely on order of items in slice

Reply:
Done on HEAD of PR. Good suggestion.